test/mcl.jl

```diff
@@ -58,3 +58,16 @@ res = mcl(diagm([1.0, 1.0]), display=:none, allow_singles=false)
 @test length(res.counts) == 0
 @test res.assignments == [0, 0]
 @test res.nunassigned == 2
+
+# use sparse matrix
+res = mcl(sparse(adj_matrix), display=:none, expansion=2)
+@test isa(res, MCLResult)
+@test length(res.assignments) == length(nodes)
+@test res.nunassigned == 0
+
+@test_throws MethodError mcl(sparse(adj_matrix), display=:none, expansion=2.0)
```

Comment:
Maybe `@test_broken` since it's nothing wrong in calling `mcl()` with these parameters, it's just that `^` is not implemented for sparse matrices yet?

Reply:
Ah, please ignore the comment above -- I've just realized it doesn't make sense to implement `^non-integer` for sparse matrices.
It could make sense to check sparse `adj` + non-integer `expansion` combination in `mcl()` and throw `ArgumentError` with a more specific description, but `MethodError` should be fine for now.

Reply:
See my above comment, better to throw earlier.